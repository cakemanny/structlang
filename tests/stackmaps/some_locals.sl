struct X { a: bool, b: bool, c: bool, d: bool, }

fn f(x: X) -> int {
    if (x.a && x.c) {
        5
    } else {
        10
    }
}

fn main() -> int {
    let x: *X = new X { false, true, false, true };
    f(*x)
}
