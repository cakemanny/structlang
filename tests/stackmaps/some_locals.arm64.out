	.section	__DATA,__const
	.p2align	3
Lptrmap0:
	.quad	0
	.quad	Lret12	; return address - the key
	.long	699050	; callee-save bitmap
	.short	2	; number of stack args + 2
	.short	2	; length of locals space
	.short	1	; length of spills space
	.byte	170	; spill_reg
	.byte	170	; spill_reg
	.byte	170	; spill_reg
	.byte	170	; spill_reg
	.byte	170	; spill_reg
	.zero	1
	.quad	0	; arg bitmap
	.quad	2	; locals bitmap
	.quad	0	; spills bitmap
	.p2align	3
Lptrmap1:
	.quad	Lptrmap0
	.quad	Lret11	; return address - the key
	.long	699050	; callee-save bitmap
	.short	2	; number of stack args + 2
	.short	2	; length of locals space
	.short	1	; length of spills space
	.byte	170	; spill_reg
	.byte	170	; spill_reg
	.byte	170	; spill_reg
	.byte	170	; spill_reg
	.byte	170	; spill_reg
	.zero	1
	.quad	0	; arg bitmap
	.quad	0	; locals bitmap
	.quad	0	; spills bitmap
	.globl	_sl_rt_frame_maps
	.p2align	3
_sl_rt_frame_maps:
	.quad	Lptrmap1
